<Type Name="ThreadExceptionEventArgs" FullName="System.Threading.ThreadExceptionEventArgs">
  <TypeSignature Language="C#" Value="public class ThreadExceptionEventArgs : EventArgs" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit ThreadExceptionEventArgs extends System.EventArgs" />
  <TypeSignature Language="DocId" Value="T:System.Threading.ThreadExceptionEventArgs" />
  <TypeSignature Language="VB.NET" Value="Public Class ThreadExceptionEventArgs&#xA;Inherits EventArgs" />
  <TypeSignature Language="C++ CLI" Value="public ref class ThreadExceptionEventArgs : EventArgs" />
  <AssemblyInfo>
    <AssemblyName>System</AssemblyName>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System.Threading.Thread</AssemblyName>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.1.1.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.EventArgs</BaseTypeName>
  </Base>
  <Interfaces />
  <Docs>
    <summary><span data-ttu-id="1570d-101">Предоставляет данные для события <see cref="E:System.Windows.Forms.Application.ThreadException" />.</span><span class="sxs-lookup"><span data-stu-id="1570d-101">Provides data for the <see cref="E:System.Windows.Forms.Application.ThreadException" /> event.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="1570d-102">Объект <xref:System.Threading.ThreadExceptionEventArgs> создается потоком при возникновении необработанного исключения.</span><span class="sxs-lookup"><span data-stu-id="1570d-102">A <xref:System.Threading.ThreadExceptionEventArgs> is created by a thread when an unhandled exception occurs.</span></span> <span data-ttu-id="1570d-103"><xref:System.Threading.ThreadExceptionEventArgs> содержит <xref:System.Exception> возникшее.</span><span class="sxs-lookup"><span data-stu-id="1570d-103"><xref:System.Threading.ThreadExceptionEventArgs> contains the <xref:System.Exception> that occurred.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="1570d-104">Следующий пример позволяет вызывать <xref:System.Windows.Forms.Application.ThreadException> событий, щелкнув `button1` в форме.</span><span class="sxs-lookup"><span data-stu-id="1570d-104">The following example allows you to raise a <xref:System.Windows.Forms.Application.ThreadException> event by clicking `button1` on a form.</span></span> <span data-ttu-id="1570d-105">В примере создается два класса.</span><span class="sxs-lookup"><span data-stu-id="1570d-105">The example creates two classes.</span></span> <span data-ttu-id="1570d-106">`ErrorHandler` Класс создает форму и кнопки, которая вызывает событие.</span><span class="sxs-lookup"><span data-stu-id="1570d-106">The `ErrorHandler` class creates the form and the button that raises the event.</span></span> <span data-ttu-id="1570d-107">`CustomExceptionHandler` Класс предоставляет методы для обработки исключения.</span><span class="sxs-lookup"><span data-stu-id="1570d-107">The `CustomExceptionHandler` class provides the methods to handle the exception.</span></span>  
  
 <span data-ttu-id="1570d-108">В `Main` в `ErrorHandler` класса, код создает новый экземпляр класса, то есть обработчика исключения экземпляр `CustomExceptionHandler`.</span><span class="sxs-lookup"><span data-stu-id="1570d-108">In `Main` in the `ErrorHandler` class, the code creates a new instance of the exception handling class, that is, an instance of the `CustomExceptionHandler`.</span></span> <span data-ttu-id="1570d-109">Затем экземпляр добавляемого события и запустите приложение.</span><span class="sxs-lookup"><span data-stu-id="1570d-109">Then the instance is added to the event, and the application is run.</span></span>  
  
 <span data-ttu-id="1570d-110">В `OnThreadException` метод в `CustomExceptionHandler` класс, используемый в примере `try...catch...finally` инструкции для обработки исключения.</span><span class="sxs-lookup"><span data-stu-id="1570d-110">In the `OnThreadException` method in the `CustomExceptionHandler` class, the example uses a `try...catch...finally` statement to process the exception.</span></span> <span data-ttu-id="1570d-111">`ShowThreadExceptionDialog` Метод создает сообщение, отображаемое и отображает его в окне сообщения.</span><span class="sxs-lookup"><span data-stu-id="1570d-111">The `ShowThreadExceptionDialog` method creates the message to display, and displays it in a message box.</span></span>  
  
 [!code-csharp[Classic ThreadExceptionEventArgs Example#1](~/samples/snippets/csharp/VS_Snippets_CLR_Classic/classic ThreadExceptionEventArgs Example/CS/source.cs#1)]
 [!code-vb[Classic ThreadExceptionEventArgs Example#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_Classic/classic ThreadExceptionEventArgs Example/VB/source.vb#1)]  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Threading.Thread" />
    <altmember cref="T:System.Threading.ThreadStart" />
    <altmember cref="T:System.Threading.ThreadExceptionEventHandler" />
  </Docs>
  <Members>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ThreadExceptionEventArgs (Exception t);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(class System.Exception t) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Threading.ThreadExceptionEventArgs.#ctor(System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (t As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ThreadExceptionEventArgs(Exception ^ t);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Thread</AssemblyName>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="t" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="t"><span data-ttu-id="1570d-112">Возникшее исключение (<see cref="T:System.Exception" />).</span><span class="sxs-lookup"><span data-stu-id="1570d-112">The <see cref="T:System.Exception" /> that occurred.</span></span></param>
        <summary><span data-ttu-id="1570d-113">Инициализирует новый экземпляр класса <see cref="T:System.Threading.ThreadExceptionEventArgs" />.</span><span class="sxs-lookup"><span data-stu-id="1570d-113">Initializes a new instance of the <see cref="T:System.Threading.ThreadExceptionEventArgs" /> class.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Examples  
 <span data-ttu-id="1570d-114">Следующий пример позволяет вызывать <xref:System.Windows.Forms.Application.ThreadException> событий, щелкнув `button1` в форме.</span><span class="sxs-lookup"><span data-stu-id="1570d-114">The following example allows you to raise a <xref:System.Windows.Forms.Application.ThreadException> event by clicking `button1` on a form.</span></span> <span data-ttu-id="1570d-115">В примере создается два класса.</span><span class="sxs-lookup"><span data-stu-id="1570d-115">The example creates two classes.</span></span> <span data-ttu-id="1570d-116">`ErrorHandler` Класс создает форму и кнопки, которая вызывает событие.</span><span class="sxs-lookup"><span data-stu-id="1570d-116">The `ErrorHandler` class creates the form and the button that raises the event.</span></span> <span data-ttu-id="1570d-117">`CustomExceptionHandler` Класс предоставляет методы для обработки исключения.</span><span class="sxs-lookup"><span data-stu-id="1570d-117">The `CustomExceptionHandler` class provides the methods to handle the exception.</span></span>  
  
 <span data-ttu-id="1570d-118">В `Main` в `ErrorHandler` класса, код создает новый экземпляр класса, то есть обработчика исключения экземпляр `CustomExceptionHandler`.</span><span class="sxs-lookup"><span data-stu-id="1570d-118">In `Main` in the `ErrorHandler` class, the code creates a new instance of the exception handling class, that is, an instance of the `CustomExceptionHandler`.</span></span> <span data-ttu-id="1570d-119">Затем экземпляр добавляемого события и запустите приложение.</span><span class="sxs-lookup"><span data-stu-id="1570d-119">Then the instance is added to the event, and the application is run.</span></span>  
  
 <span data-ttu-id="1570d-120">В `OnThreadException` метод в `CustomExceptionHandler` класс, используемый в примере `try...catch...finally` инструкции для обработки исключения.</span><span class="sxs-lookup"><span data-stu-id="1570d-120">In the `OnThreadException` method in the `CustomExceptionHandler` class, the example uses a `try...catch...finally` statement to process the exception.</span></span> <span data-ttu-id="1570d-121">`ShowThreadExceptionDialog` Метод создает сообщение, отображаемое и отображает его в окне сообщения.</span><span class="sxs-lookup"><span data-stu-id="1570d-121">The `ShowThreadExceptionDialog` method creates the message to display, and displays it in a message box.</span></span>  
  
 [!code-csharp[Classic ThreadExceptionEventArgs Example#1](~/samples/snippets/csharp/VS_Snippets_CLR_Classic/classic ThreadExceptionEventArgs Example/CS/source.cs#1)]
 [!code-vb[Classic ThreadExceptionEventArgs Example#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_Classic/classic ThreadExceptionEventArgs Example/VB/source.vb#1)]  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Threading.Thread" />
        <altmember cref="T:System.Threading.ThreadStart" />
      </Docs>
    </Member>
    <Member MemberName="Exception">
      <MemberSignature Language="C#" Value="public Exception Exception { get; }" />
      <MemberSignature Language="ILAsm" Value=".property instance class System.Exception Exception" />
      <MemberSignature Language="DocId" Value="P:System.Threading.ThreadExceptionEventArgs.Exception" />
      <MemberSignature Language="VB.NET" Value="Public ReadOnly Property Exception As Exception" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; property Exception ^ Exception { Exception ^ get(); };" />
      <MemberType>Property</MemberType>
      <AssemblyInfo>
        <AssemblyName>System</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Threading.Thread</AssemblyName>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Exception</ReturnType>
      </ReturnValue>
      <Docs>
        <summary><span data-ttu-id="1570d-122">Получает тип <see cref="T:System.Exception" /> произошедшего события.</span><span class="sxs-lookup"><span data-stu-id="1570d-122">Gets the <see cref="T:System.Exception" /> that occurred.</span></span></summary>
        <value><span data-ttu-id="1570d-123">Возникшее исключение (<see cref="T:System.Exception" />).</span><span class="sxs-lookup"><span data-stu-id="1570d-123">The <see cref="T:System.Exception" /> that occurred.</span></span></value>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Examples  
 <span data-ttu-id="1570d-124">Следующий пример позволяет вызывать <xref:System.Windows.Forms.Application.ThreadException> событий, щелкнув `button1` в форме.</span><span class="sxs-lookup"><span data-stu-id="1570d-124">The following example allows you to raise a <xref:System.Windows.Forms.Application.ThreadException> event by clicking `button1` on a form.</span></span> <span data-ttu-id="1570d-125">В примере создается два класса.</span><span class="sxs-lookup"><span data-stu-id="1570d-125">The example creates two classes.</span></span> <span data-ttu-id="1570d-126">`ErrorHandler` Класс создает форму и кнопки, которая вызывает событие.</span><span class="sxs-lookup"><span data-stu-id="1570d-126">The `ErrorHandler` class creates the form and the button that raises the event.</span></span> <span data-ttu-id="1570d-127">`CustomExceptionHandler` Класс предоставляет методы для обработки исключения.</span><span class="sxs-lookup"><span data-stu-id="1570d-127">The `CustomExceptionHandler` class provides the methods to handle the exception.</span></span>  
  
 <span data-ttu-id="1570d-128">В `Main` в `ErrorHandler` класса, код создает новый экземпляр класса, то есть обработчика исключения экземпляр `CustomExceptionHandler`.</span><span class="sxs-lookup"><span data-stu-id="1570d-128">In `Main` in the `ErrorHandler` class, the code creates a new instance of the exception handling class, that is, an instance of the `CustomExceptionHandler`.</span></span> <span data-ttu-id="1570d-129">Затем экземпляр добавляемого события и запустите приложение.</span><span class="sxs-lookup"><span data-stu-id="1570d-129">Then the instance is added to the event, and the application is run.</span></span>  
  
 <span data-ttu-id="1570d-130">В `OnThreadException` метод в `CustomExceptionHandler` класс, используемый в примере `try...catch...finally` инструкции для обработки исключения.</span><span class="sxs-lookup"><span data-stu-id="1570d-130">In the `OnThreadException` method in the `CustomExceptionHandler` class, the example uses a `try...catch...finally` statement to process the exception.</span></span> <span data-ttu-id="1570d-131">`ShowThreadExceptionDialog` Метод создает сообщение, отображаемое и отображает его в окне сообщения.</span><span class="sxs-lookup"><span data-stu-id="1570d-131">The `ShowThreadExceptionDialog` method creates the message to display, and displays it in a message box.</span></span>  
  
 [!code-csharp[Classic ThreadExceptionEventArgs Example#1](~/samples/snippets/csharp/VS_Snippets_CLR_Classic/classic ThreadExceptionEventArgs Example/CS/source.cs#1)]
 [!code-vb[Classic ThreadExceptionEventArgs Example#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_Classic/classic ThreadExceptionEventArgs Example/VB/source.vb#1)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>